[dv] Document how to reproduce CI (Private) failure (#5849)
lowRISC/opentitan: doc/ug/getting_started_dv.md

The [UART DV](https://github.com/lowRISC/opentitan/tree/master/hw/ip/uart/dv) area can be used as a canonical example for making progress.
If it is not clear on how to proceed, feel free to file an issue requesting assistance.

> ## Reproduce DV failure in CI(private)

Reviewer: ```suggestion
## Reproduce a DV failure in CI
```

(This is true for all kinds of DV CI failures.)

> Follow these steps to reproduce CI(private) failure

Reviewer 2: ```suggestion
Follow these steps to reproduce failures in private CI.
```

1. Make sure the version of VCS is the same as the [one](https://github.com/lowRISC/opentitan-private-ci/blob/master/jobs.yml#L5) running in CI

> 2. CI runs against an auto-generated merge commit, which effectively is the PR merge into master.

Reviewer 2: ```suggestion
2. CI runs against an auto-generated merge commit, which effectively is the pull request (PR) merged into master.
```

Reviewer 3: As Sri said, or maybe this would be clearer as something like "..., which is generated by merging the PR into the master branch"?

> This "merge" branch gets updated automatically by GitHub whenever you push to the PR branch, or when you close/re-open the PR.

Reviewer 2: ```suggestion
This "merge" branch is updated automatically by GitHub whenever the PR branch is pushed, or when the PR is (re)opened or closed.
```

> Get the extra database by doing the following (Assuming "upstream" is the name of your lowrisc/opentian repository)

Reviewer 2: ```suggestion
Retrieve this database by running (assuming "upstream" is the name of your lowrisc/opentian repository):
```

Reviewer: Can we avoid the term "database"? It always confuses me when I read it -- a directory full of files is just not a database in the general use of the term.

```console

> $ git fetch upstream pull/<PR_number>/merge:<your_branch>

Reviewer 2: Perhaps `s/your_branch/temp_branch`?
For people like me who are (still :-/) confused by Git, also clarify here this has nothing to do with the branch that was originally pushed to open the PR.

Reviewer 3: Maybe it would be better to split this differently?
```
git fetch upstream pull/<PR_number>/merge
git checkout -b <your_branch> FETCH_HEAD
```
I think that probably makes it clearer what's going on (and is what I do here!)

$ git checkout <your_branch>
```

3. This is the command that CI runs for the smoke regression.
You can use it or only run the failed test with `--fixed-seed=1` to reproduce the failure
```console
$ util/dvsim/dvsim.py hw/top_earlgrey/dv/top_earlgrey_sim_cfgs.hjson -i smoke --fixed-seed=1

> ```

Reviewer 2: Add step to run a specific failing test as well. Add `[--waves]` to the command line to indicate how to dump waves optionally.

The [UART DV](https://github.com/lowRISC/opentitan/tree/master/hw/ip/uart/dv) area can be used as a canonical example for making progress.
If it is not clear on how to proceed, feel free to file an issue requesting assistance.

## Reproduce a DV failure in CI

Follow these steps to reproduce the failure

1. Make sure the version of VCS is the same as the [one](https://github.com/lowRISC/opentitan-private-ci/blob/master/jobs.yml#L5) running in CI.

> 2. CI runs against an auto-generated merge commit, which effectively is generated by merging the pull request (PR) into the master branch.

Reviewer 4: sorry this is not related to this PR exactly, but i'm sort of confused by this.
Does this mean what CI runs is not the same as what I have locally when I push the PR?

Because when it says "merged" into master, that almost makes me think it gets rebased on to tip of tree or something.

Author: right, very likely what CI runs is not the same as you local has. it gets rebased on to tip of tree and I think that makes more sense. Your local branch can be very old. I didn't know it until Philipp told me.
